pwndbg/commands/context.py

```diff
@@ -472,6 +472,15 @@ def get_regs(*regs):
 code_lines = pwndbg.config.Parameter('context-code-lines', 10, 'number of additional lines to print in the code context')
 
 def context_disasm(target=sys.stdout, with_banner=True, width=None):
+    flavor = gdb.execute('show disassembly-flavor', to_string=True).lower().split('"')[1]
+    syntax = pwndbg.disasm.CapstoneSyntax[flavor]
+
+    cache_get_disassembler_cached = pwndbg.memoize.reset_on_objfile(pwndbg.disasm.get_disassembler_cached)
+    cs = list(cache_get_disassembler_cached.__dict__['cache'].values())[-1]
```

Comment:
Does this really work? The `reset_on_objfile` is a class and the above line will create an instance, so it should have `cache` field re-created each time we execute this?

Reply:
I wonder if we can overwriite GDB parameter with our own, and just clear caches if its being set...

Reply:
> Does this really work? The `reset_on_objfile` is a class and the above line will create an instance, so it should have `cache` field re-created each time we execute this?

Yes, I tested on Dockerfile provided in your project and I modified these lines in my Arch Linux with the package installed `pacman -S pwndbg` (community/pwndbg 2020.07.23-2), and it's working.

No, it didn't recreate every time, I'm prevented this comparing the cache value for CS.syntax versus the GDB flavor, means it only re-create the cache every time it's different, means every time you change manually.

I know it look a dirty solution but I didn't find other way, I have some ideas but I'll share in the next comment.

Reply:
> I wonder if we can overwriite GDB parameter with our own or just hook one, and just clear caches if its being set...

It' sounds good, create the pwndbg command, the only missing part will be when the user set the flavor in the `.gdbinit`, just this file is the first and the only time that the flavor is created, except if the caches are cleaned. Also, it will be noisy that I have this option but I need to set the flavor manually.

Let me create a YouTube video and show how these changes are working.

Reply:
> Let me create a YouTube video and show how these changes are working.

Thanks <3. Fwiw there's also https://asciinema.org/

Reply:
> > Let me create a YouTube video and show how these changes are working.
> 
> Thanks <3. Fwiw there's also https://asciinema.org/

Awesome, I tried and it's an excellent idea, record the STD Output and record the time waiting from one screen to another.

Reply:
I had some time to revisit this today and, AWESOME video, thanks!

My initial concern was more about that we call `reset_on_objfile` here: it seems unnecessary?

It seems we should be able to do just this:
```py
# Get the Capstone object to set disassembly syntax
cs = next(iter(pwndbg.disasm.get_disassembler_cached.cache.values()))
```

This way, we don't create the `reset_on_objfile` object/instance and also a list (since we do `next(iter(<dict-values object>))`).

Can you check if that works fine? (maybe there's a reason I miss we need to create the memoize object)
If that works fine, I think we can merge this.

For other of your comments: I think making a command like `gdb flavor <flavor>` is not good as people would rather use the default/standard `set disassembly-flavor <...>`.

Also, the `pwndbg.config.Parameter` class we have, makes so we can add additonal "gdb parameters" that can be set/shown using the standard GDB's `set ...` or `show ...` command. However, I think that if we try to overwrite GDB's original parameter this way, we won't be able to use/set the original's value: so I think overwriting it may not be the best idea. Sooo I would just make it as it is in this PR :).

Reply:
You created an AWESOME app, thanks for it.

Your solution is beautiful, much better than my implementation. I applied and It works. I'll update the PR.

Yes, I'm totally agreed the other solutions not looks good, but this solution is working and keeps the code quality high.